presto-native-execution/presto_cpp/main/common/tests/ConfigTest.cpp

```diff
@@ -255,4 +256,25 @@ TEST_F(ConfigTest, optionalNodeId) {
   EXPECT_EQ(nodeId, config.nodeId());
 }
 
+TEST_F(ConfigTest, readConfigEnvVarTest) {
+  setUpConfigFilePath();
+  std::string ENV_VAR = "PRESTO_READ_CONFIG_TEST_VAR";
+
+  writeConfigFile(
+      fmt::format("{}={}\n", "plain-text", "plain-text-value") +
+      fmt::format("{}=${{{}}}\n", "env-var", ENV_VAR) +
+      fmt::format("{}=${{{}\n", "env-var2", ENV_VAR) +
+      fmt::format("{}={}}}\n", "env-var3", ENV_VAR) +
+      fmt::format("{}=${{}}\n", "no-env-var"));
+
+  setenv(ENV_VAR.c_str(), "env-var-value", 1);
```

Comment:
Add a test for empty value. It should be enough to check that it does not introduce a crash.